### ISSUE TITLE
Remove artifact from server monitoring table

### ISSUE THREAD

**Issue by marianoka** [Velocidex/velociraptor#1181]
Hi!
how do I manage to stop or remove an artifact from sever monitoring table? Is is posible?

**Comment by scudette** [comment]
Just unselect it from the GUI and click the save button? Is this not
working?

**Comment by marianoka** [comment]
At least, in my installation is not. In fact, I don't see a save button. My steps are:
1-Go to server events
2-Click Update server monitoriing table
3-Unselect all
4- press lunch
But After that I still see the same artifacts.
Thanks

**Comment by scudette** [comment]
You can remove them from the table in either the artifact selection step or the parameters step:
![Velociraptor_Response_and_Monitoring](https://user-images.githubusercontent.com/3856546/128701769-21bd087e-dd23-4adb-8493-fd1df1222caf.png)

When the monitoring table in review step and it should be the same as the one you see in the "Raw Server Monitoring Table JSON" view

Which artifact are you trying to remove? How do you know it is not working?

**Comment by marianoka** [comment]
![imagen](https://user-images.githubusercontent.com/9213379/128702208-a5c8a06b-9e51-4e26-bfa2-b6b5df65937b.png)
I Have no artifacts but I see this
![imagen](https://user-images.githubusercontent.com/9213379/128702263-687784e0-654b-41fe-83d1-a0394d20ee29.png)

**Comment by scudette** [comment]
The artifacts you see under the "Select Artifacts" pull down are the ones that have some data that you can view. It does not mean they are actually being currently collected - just that they collected data in the past sometime and you can see it if you like. You can see the currently collecting artifacts by clicking the "Raw Server Monitoring Table JSON" button 

![Velociraptor_Response_and_Monitoring](https://user-images.githubusercontent.com/3856546/128702528-66875b59-bc82-4d06-bf16-7cf776d3744b.png)

For example this might look like this:

![Velociraptor_Response_and_Monitoring](https://user-images.githubusercontent.com/3856546/128702698-51c05621-baf5-4421-91ab-e94ba8ebbf6f.png)

**Comment by marianoka** [comment]
Txh for you answer. I have checked that option, and even there is no artifact active, I have a custom artifact generating logs.
![imagen](https://user-images.githubusercontent.com/9213379/128702990-9c10708f-eeb9-43a3-83c4-3579a918e69d.png)
![imagen](https://user-images.githubusercontent.com/9213379/128703015-ac678dc0-64d6-4f40-979d-06a28046d674.png)

**Comment by scudette** [comment]
Interesting! 

These logs seem to be from 12 hours ago - was that artifact collecting then?  Are there any more recent logs?

**Comment by marianoka** [comment]
Yes. I have a lot of logs. I will try deleting the artifact and recreating

**Comment by scudette** [comment]
Can you try to just set the Server.Monitoring.Health artifact? Normally we always collect that one so the dashboard continues working - maybe it doesnt like having a totally empty table

**Comment by marianoka** [comment]
Fine. It's done.

**Comment by marianoka** [comment]
well. I have deleted and recreated and now I cannot even add the artifact for server monitoring. 
![imagen](https://user-images.githubusercontent.com/9213379/128708553-222b535f-2444-42f6-9034-4a7a98457305.png)
and at server monitoring Is not available.
![imagen](https://user-images.githubusercontent.com/9213379/128708660-ed5def41-0b85-449d-910d-ad4b3cf552a3.png)

**Comment by scudette** [comment]
Make sure you set the artifact type as SERVER_EVENT otherwise it is not visible in the server monitoring GUI

There does seem to be a problem with setting an empty table. I am looking into it now. The issue seems to be that the running queries are not stopped when the new table is totally empty. Restarting the server actually fixes the issue because Velociraptor does not like an empty table either - it will initialize it to the Server.Monitor.Health default one.

Probably we should not allow that artifact to be removed from the table.

**Comment by marianoka** [comment]
Thanks
was an indentation problem

**Comment by marianoka** [comment]
By the way, is possible to debug an custom artifact?

**Comment by scudette** [comment]
The best way to develop the query is in the notebook - just add lots of print messages using the log() vql function. You can also have log messages in the artifact but it is a bit more tedious to iterate.

See the section on log messages here https://docs.velociraptor.app/docs/vql/#lazy-evaluation and follow the videos in the training course.
